packages/contracts-bedrock/scripts/go-ffi/go-ffi.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```

Comment:
I added a couple comments to this file. As a high level I'm wondering if the logic can be rewritten as: 

- if the binary exists:
    - run the binary and exit
  - else:
    - check for the lockfile
    - if the lockfile exists
	    - wait for the binary to exist
	    - once it does run the binary and exit
    - else:  	
		- create the lock file
		- build the binary
		- remove the lockfile
		- run the binary and exit

Reply:
`run the binary and exit` is the common part, so I put it in the end to reduce code.